client/protocol.go

```diff
@@ -424,23 +431,43 @@ OuterLoop:
 				}
 				okCnt++
 			}
-
 			if okCnt >= majorityCnt {
-				break OuterLoop
+				break
 			}
-		case err := <-errCh:
-			c.logger.Warn("fetch cluster error", zap.Error(err))
 		}
+
+		c.logger.Info("fetch cluster succeeded")
+		c.state.leader = *res.LeaderId
+		c.state.term = res.Term
+		c.state.checkAndUpdate(*res.LeaderId, res.Term)
```

Comment:
We should return here if the fetch is successful, otherwise it will continue to retry.

Reply:
Done.